src/main/java/io/openliberty/tools/common/plugins/util/DevUtil.java

```diff
@@ -5696,27 +5703,35 @@ private Set<String> getGeneratedFeatures() {
     public void writeDevcMetadata(boolean alive) {
         File metaFile = new File(buildDirectory, serverDirectory.getName() + "-liberty-devc-metadata.xml");
         try {
-            XMLStreamWriter metadataWriter = XMLOutputFactory.newInstance().createXMLStreamWriter(new FileWriter(metaFile)) ;
-            metadataWriter.writeStartDocument();
-            metadataWriter.writeStartElement("devcModeMetaData");
-            writeElement(metadataWriter, "containerEngine", isDocker ? DEVC_CONTAINER_DOCKER : DEVC_CONTAINER_PODMAN);
-            writeElement(metadataWriter, "containerName", containerName != null ? containerName : DEVMODE_CONTAINER_BASE_NAME);
-            writeElement(metadataWriter, "imageName", imageName);
-            if (containerfile != null) {
-                writeElement(metadataWriter, "containerfile", containerfile.getCanonicalPath());
-            }
-            if (containerBuildContext != null) {
-                writeElement(metadataWriter, "containerBuildContext", containerBuildContext.getCanonicalPath());
-            }
-            writeElement(metadataWriter, "containerAlive", String.valueOf(alive));
-            writeElement(metadataWriter, "containerBuildTimeout", Integer.toString(containerBuildTimeout));
-            writeElement(metadataWriter, "containerRunOpts", containerRunOpts);
-            metadataWriter.writeEndElement();
-            metadataWriter.writeEndDocument();
-            metadataWriter.flush();
-            metadataWriter.close();
+            FileWriter metaFileWriter = new FileWriter(metaFile);
+            XMLStreamWriter metadataWriter = XMLOutputFactory.newInstance().createXMLStreamWriter(metaFileWriter) ;
```

Comment:
There is the slight possibility that this could leave open the `FileWriter` if there is an exception when creating the `XMLStreamWriter`. Should just declare these vars and set to null here, then create inside the `try` that has the `finally`. In the `finally`, only call flush/close if they are not null.